src/AutoRest.CSharp/AutoRest/Plugins/CSharpProj.cs

```diff
@@ -31,7 +33,21 @@ internal class CSharpProj : IPlugin
         public async Task<bool> Execute(IPluginCommunication autoRest)
         {
             var ns = await autoRest.GetValue<string>("namespace");
-            await autoRest.WriteFile($"{Configuration.ProjectRelativeDirectory}{ns}.csproj", _csProjContent, "source-file-csharp");
+
+            string codeModelFileName = (await autoRest.ListInputs()).FirstOrDefault();
+            if (string.IsNullOrEmpty(codeModelFileName))
+                throw new Exception("Generator did not receive the code model file.");
+
+            Task<CodeModel> codeModelTask = Task.Run(async () =>
```

Comment:
You don't need the Task.Run here, you are not doing anything in parallel with it.